exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java

```diff
@@ -55,8 +56,12 @@ public static PrometheusHttpServerBuilder builder() {
       int port,
       @Nullable ExecutorService executor,
       PrometheusRegistry prometheusRegistry,
-      boolean otelScopeEnabled) {
-    this.prometheusMetricReader = new PrometheusMetricReader(otelScopeEnabled);
+      boolean otelScopeEnabled,
+      boolean addResourceAttributesAsLabels,
+      Predicate<String> allowedResourceAttributesFilter) {
```

Comment:
The same information which is conveyed with two properties here can be conveyed with a `@Nullable Predicate<String>`:
- null is equal to `addResourceAttributeAsLabels = false`
- non-null is equal to `addResourceAttributesAsLabels = true`

Generally we like to avoid `@Nullable` where we can, but in this case, I think its preferable because its easy to have two properties which disagree with each other. I.e. I have a predicate that says include all the attributes but accidentally set `addResourceAttributesAsLabels = false`. 

Also, the `@Nullable` should be an implementation detail - not something exposed in the user facing API.

This advice to use a single `@Nullable Predicate<String>` applies in several places where this pair of properties currently appears.

Reply:
You are correct. I thought that using @Nullable is not recommended, which is why I followed that route. I will fix it now.

Reply:
I think the maintainers are generally in agreement in "avoiding null when you can", but that there are some situations where avoiding it is just too unnatural. We might not always be consistent on where we draw the line, but for me, I try much harder to avoid `null` if its public API surface area we're talking about. This all being package private, I'm less interested in finding a creative solution that avoids null.